apache/incubator-zeppelin
# Auto-suggestion feature for notebook permissions

### What is this PR for?
This PR will provide user-friendly way to  give notebook permissions. It will show suggestions on the basis of usernames matching the input characters.
It will  communicate with Shiro to fetch usernames from the configured realms and select usernames matching the input character and shows it to the user as suggestions.
Currently, it works with shiro.ini file, LDAP, JDBC. In future we can extend it to work with other realms too.

### What type of PR is it?
Improvement

### Todos


### What is the Jira issue?
[ZEPPELIN-933](https://issues.apache.org/jira/browse/ZEPPELIN-933)
### How should this be tested?
1. Authenticate Zeppelin with any of the realms mention above(LDAP, JDBC,ini).
2. In notebook permission section, try giving permissions to some notebook and press some initial characters . It should show matching users as suggestions and you should be able to select users from the suggestions.
3. You should be able to give usernames separated by comma and finally save it.


### Screenshots (if appropriate)
![screen shot 2016-06-03 at 5 56 01 pm](https://cloud.githubusercontent.com/assets/7026661/15778542/8a208492-29b4-11e6-8bf9-3a31ce90223a.png)




### Questions:
* Does the licenses files need update?no
* Is there breaking changes for older versions?no
* Does this needs documentation?maybe
